Source/Core/CoplanarPolygonGeometryLibrary.js

```diff
@@ -0,0 +1,131 @@
+        Check.defined('positions', positions);
+        Check.defined('positionsResult', positionsResult);
+        //>>includeEnd('debug');
+
+        var obb = OrientedBoundingBox.fromPoints(positions, obbScratch);
+        var halfAxes = obb.halfAxes;
+        var xAxis = Matrix3.getColumn(halfAxes, 0, scratchXAxis);
+        var yAxis = Matrix3.getColumn(halfAxes, 1, scratchYAxis);
+        var zAxis = Matrix3.getColumn(halfAxes, 2, scratchZAxis);
+
+        var xMag = Cartesian3.magnitude(xAxis);
+        var yMag = Cartesian3.magnitude(yAxis);
+        var zMag = Cartesian3.magnitude(zAxis);
+        var min = Math.min(xMag, yMag, zMag);
+
+        var i;
+        // If all the points are on a line, just remove one of the zero dimensions
+        if ((xMag === 0 && (yMag === 0 || zMag === 0)) || (yMag === 0 && zMag === 0)) {
+            return;
+        }
+        var center = obb.center;
+        var planeXAxis;
+        var planeYAxis;
+        var origin = Cartesian3.clone(center, scratchOrigin);
+        var normal;
+        if (min === xMag) {
+            if (min !== 0) {
+                origin = Cartesian3.add(origin, xAxis, origin);
+                normal = Cartesian3.normalize(xAxis, scratchNormal);
+            }
+            planeXAxis = Cartesian3.normalize(yAxis, yAxis);
+            planeYAxis = Cartesian3.normalize(zAxis, zAxis);
+        } else if (min === yMag) {
+            if (min !== 0) {
+                origin = Cartesian3.add(origin, yAxis, origin);
+                normal = Cartesian3.normalize(yAxis, scratchNormal);
+            }
+            planeXAxis = Cartesian3.normalize(xAxis, xAxis);
+            planeYAxis = Cartesian3.normalize(zAxis, zAxis);
+        } else {
+            if (min !== 0) {
+                origin = Cartesian3.add(origin, zAxis, origin);
+                normal = Cartesian3.normalize(zAxis, scratchNormal);
+            }
+            planeXAxis = Cartesian3.normalize(xAxis, xAxis);
+            planeYAxis = Cartesian3.normalize(yAxis, yAxis);
+        }
+
+        if (min === 0) {
+            normal = Cartesian3.cross(planeXAxis, planeYAxis, scratchNormal);
+            normal = Cartesian3.normalize(normal, normal);
+        }
+
+        if (defined(normalResult)) {
+            Cartesian3.clone(normal, normalResult);
+        }
+        if (defined(tangentResult)) {
+            Cartesian3.clone(planeXAxis, tangentResult);
+        }
+        if (defined(bitangentResult)) {
+            Cartesian3.clone(planeYAxis, bitangentResult);
+        }
+
+        var plane = Plane.fromPointNormal(origin, normal, scratchPlane);
+        var ray = scratchRay;
+        ray.direction = Cartesian3.clone(normal, ray.direction);
+
+        for (i = 0; i < positions.length; i++) {
+            ray.origin = Cartesian3.clone(positions[i], ray.origin);
+
+            var intersectionPoint = IntersectionTests.rayPlane(ray, plane, scratchIntersectionPoint);
```

Comment:
You don't need to intersect the ray with the plane. You just need this:
```javascript
v = subtract(positions[i], origin);
x = dot(xAxis, v);
y = dot(yAxis, v);
```

Reply:
I agree that's the case when the positions are actually coplanar.  But I need it to work with positions that are mostly coplanar and to "try it's best" for a set of arbitrary points so I think I still need this?

Reply:
Hmm, I guess you're right though, the results come out the same for both. Thanks!